VideoLocker/src/main/java/org/edx/mobile/services/CourseManager.java

```diff
@@ -0,0 +1,183 @@
+package org.edx.mobile.services;
+
+import android.text.TextUtils;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+
+import org.edx.mobile.model.ChapterModel;
+import org.edx.mobile.model.CourseModel;
+import org.edx.mobile.model.IChapter;
+import org.edx.mobile.model.ICourse;
+import org.edx.mobile.model.ISequential;
+import org.edx.mobile.model.IVertical;
+import org.edx.mobile.model.SequentialModel;
+import org.edx.mobile.model.VerticalModel;
+import org.edx.mobile.model.api.EnrolledCoursesResponse;
+import org.edx.mobile.model.api.PathModel;
+import org.edx.mobile.model.api.SummaryModel;
+import org.edx.mobile.model.api.VideoResponseModel;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * no matter it is a smart phone, phablet, or tablet, app should show
+ * the detail of only one course detail. so keep the logic at service
+ * level is better than passing around between activities and fragments.
+ *
+ * TODO - the data will be persistent.
+ *
+ */
+public class CourseManager {
+
+    private static CourseManager instance;
+
+    public static synchronized CourseManager getSharedInstance(){
+        if ( instance == null )
+            instance = new CourseManager();
+        return instance;
+    }
+
+    private ICourse courseInView;
+    private EnrolledCoursesResponse courseData;
+    private IChapter chapterInView;
+    private ISequential sequentialInView;
+
+    private CourseManager(){
+
+    }
+
+    public void clear(){
+        courseInView = null;
+        courseData = null;
+        chapterInView = null;
+        sequentialInView = null;
+    }
+
+    public ICourse getCourseInView() {
+        return courseInView;
+    }
+
+    public void setCourseInView(ICourse courseInView) {
+        this.courseInView = courseInView;
```

Comment:
Is it possible to have two courses in the stack at once? Like if you're looking at a course and then a notifications fires for a different course?